integreat_cms/static/src/js/pois/poi-actions.ts

```diff
@@ -0,0 +1,85 @@
+import { getCsrfToken } from "../utils/csrf-token";
+
+let initialLongitude: string, initialLatitude: string;
+let userInputLng: string, userInputLat: string;
+
+window.addEventListener("load", () => {
+    document.getElementById("id_address")?.addEventListener("keyup", fetchCoordinates);
+    document.getElementById("id_postcode")?.addEventListener("keyup", fetchCoordinates);
+    document.getElementById("id_city")?.addEventListener("keyup", fetchCoordinates);
+
+    document.getElementById("auto-fill-coordinates")?.addEventListener("input", fetchCoordinates);
+
+    initialLongitude = ( document.getElementById("id_longitude") as HTMLInputElement)?.value;
+    initialLatitude = (document.getElementById("id_latitude") as HTMLInputElement)?.value;
+
+    document.getElementById("id_longitude").addEventListener("keyup", saveActualCoordinates);
+    document.getElementById("id_latitude").addEventListener("keyup", saveActualCoordinates);
+});
+
+function saveActualCoordinates(){
+    userInputLng = (document.getElementById("id_longitude") as HTMLInputElement).value;
+    userInputLat = (document.getElementById("id_latitude") as HTMLInputElement).value;
+}
+
+async function fetchCoordinates() {
+    if ( ( document.getElementById("auto-fill-coordinates") as HTMLInputElement).checked ){
+        let url = "/ajax/coordinates/";
+
+        let street = ( document.getElementById("id_address") as HTMLInputElement).value;
+        let postcode= ( document.getElementById("id_postcode") as HTMLInputElement).value;
+        let city= ( document.getElementById("id_city") as HTMLInputElement).value;
+
+        if (street.trim().length == 0 || postcode.trim().length == 0 || city.trim().length == 0) {
+            return;
+        }
+
+        const response = await fetch(url, {
+            method: "POST",
+            headers: {
+                "X-CSRFToken": getCsrfToken(),
+            },
+            body: JSON.stringify({
+                street: street,
+                postcode: postcode,
+                city: city,
+            }),
+        });
+
+        if (response.status != 200) {
+            return;
+        }
+
+        const data = await response.json();
+
+        if (data) {
+            // Set and display new data
+            let longitude = data.longitude;
+            let latitude = data.latitude;
+
+            ( document.getElementById("id_longitude") as HTMLInputElement).value = longitude;
+            ( document.getElementById("id_latitude") as HTMLInputElement).value = latitude; 
+        }
```

Comment:
```suggestion
        // Set and display new data
        longitude.value = data.longitude;
        latitude.value = data.latitude;
        longitude.classList.add("!border-green-500");
        latitude.classList.add("!border-green-500");
        setTimeout(function() {
            longitude.classList.remove("!border-green-500");
            latitude.classList.remove("!border-green-500");
        }, 5000);
```
We could also show a green border after fetching to indicate the side effect